src/commands/filter_parser.cc

```diff
@@ -440,21 +433,54 @@ inline std::unique_ptr<query::Predicate> MayNegatePredicate(
 absl::StatusOr<std::unique_ptr<query::Predicate>> FilterParser::WrapPredicate(
     std::unique_ptr<query::Predicate> prev_predicate,
     std::unique_ptr<query::Predicate> predicate, bool& negate,
-    query::LogicalOperator logical_operator) {
+    query::LogicalOperator logical_operator, bool no_prev_grp,
+    bool not_rightmost_bracket) {
```

Comment:
Had a discussion about `not_rightmost_bracket`. This will be removed once Nivesh looks into this more and if we find out it is not needed